API-9080 Add require_every_value to list_archives and reports filters (#983)
livechat/livechat-public-docs: content/messaging/agent-chat-api/v3.4/index.mdx

| `sort_order` **\*\*\*\*\***                                      | No       | `string`  | Default: `desc`                                                                                                       |
| `limit`                                                          | No       | `number`  | Default: 10, min: 1, max: 100                                                                                         |
| `highlights`                                                     | No       | `object`  | Use it to highlight the match of `filters.query`. To enable highlights with default parameters, pass an empty object. |
| `highlights.pre_tag`                                             | No       | `string`  | An HTML tag to use for highlighting the matched text; default: `<em>`. Use it together with `highlights.post_tag`.    |
| `highlights.post_tag`                                            | No       | `string`  | An HTML tag to use for highlighting the matched text; default: `</em>`. Use it together with `highlights.pre_tag`.    |

**\*)**
`<properties_filter_type>` can take the following values:

- `exists` (`bool`)

> - `values` (`type[]` - an array with aspecific type for property: `string`, `int` or `bool`)

Reviewer: ```suggestion
- `values` (`type[]` - an array with a specific type for property: `string`, `int` or `bool`)
```

There's only one value allowed for a single property.

**\*\*)**
`<filter_type>` can take the following values:

- `exists` (`bool`)

> - `values` (`type[]` - an array with aspecific type for property: `string`, `int` or `bool`)

Reviewer: ```suggestion
- `values` (`type[]` - an array with a specific type for property: `string`, `int` or `bool`)
```

> - `exclude_values` (`type[]` - array with specific type for property: `string`, `int` or `bool`)

Reviewer: ```suggestion
- `exclude_values` (`type[]` - array with a specific type for property: `string`, `int` or `bool`)
```

> - `require_every_value` (`bool`) - if set to `true` will return chats that have all elements of `values` or `exclude_values` in the field that the filter applies to

Reviewer: ```suggestion
- `require_every_value` (`bool`) - if set to `true`, will return only those chats that have all elements passed in `values` or `exclude_values`
```

> Only one of `exists`, `values` or `exclude_values` may be passed.

Reviewer: ```suggestion
You can pass only one of the following values at a time: `exists`, `values` or `exclude_values`.
```

**\*\*\*)**
`<survey>` contains the following fields:

- `type` (`string`) - allowed values: `pre_chat`, `post_chat`
- `answer_id` (`string`)

**\*\*\*\*)**
`<event_type_filter_type>` can take the following values:

- `values` (`string[]` - an array of [Event](#events) types, duplicates are ignored)
- `exclude_values` (`string[]` - an array of [Event](#events) types, duplicates are ignored)

> - `require_every_value` (`bool`) - if set to `true` will return chats that have all elements of `values` or `exclude_values` in the field that the filter applies to

Reviewer: ```suggestion
- `require_every_value` (`bool`) - if set to `true`, will return only those chats that have all elements passed in `values` or `exclude_values`
```

> Only one of `values` or `exclude_values` may be passed.

Reviewer: ```suggestion
You can pass only one of the following values at a time: `values` or `exclude_values`.
```
